README.md

```diff
@@ -184,6 +184,51 @@ for item in repo_files:
     http_upload_file(repo_id=repo_id, repo_type=repo_type, file_path=item, endpoint=endpoint, token=token)
 ```
 
+### Upload to new repo or branch
+
+Before starting, please make sure you have Git-LFS installed (see [here](https://git-lfs.github.com/) for installation instructions).
+
+```python
+from pycsghub.repository import Repository
+
+token = "your access token"
+
+r = Repository(
+    repo_id="wanghh2003/ds12",
+    work_dir="/Users/hhwang/temp/ccc",
+    user_name="wanghh2003",
+    token=token,
+    repo_type="dataset"
+)
+
+# push to a new repo
+r.upload_as_new_repo(
```

Comment:
it's better to use one single method `upload`, which create repo and branch internally if needed. 
param `auto_create_repo` can be used to control repo creation. I suggest it defaults to `true`.

Reply:
Merge to one method upload.